File: app/src/main/java/com/amey/pioneerapplication/viewmodel/CategoriesViewModel.java
package com.amey.pioneerapplication.viewmodel;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

import com.amey.pioneerapplication.model.CategoriesModelItem;
import com.amey.pioneerapplication.repository.RemoteDataSource;
import com.amey.pioneerapplication.utils.AppConstants;

import java.util.List;

public class CategoriesViewModel extends ViewModel {


    RemoteDataSource remoteDataSource;
    LiveData<List<CategoriesModelItem>> categoriesModelItems;
    // TODO: Implement the ViewModel


    public void init(){
        if(categoriesModelItems != null)
            return;


        remoteDataSource = RemoteDataSource.Companion.newInstance();
        categoriesModelItems = remoteDataSource.getCategories(AppConstants.CUSTOMER_ID, AppConstants.SECRET_TOKEN);
    }

    public LiveData<List<CategoriesModelItem>> getCategories(){
        return categoriesModelItems;
    }
}

File: app/src/main/java/com/amey/pioneerapplication/viewmodel/CategoriesViewModel.kt
package com.amey.pioneerapplication.viewmodel

import androidx.lifecycle.LiveData
import androidx.lifecycle.ViewModel
import com.amey.pioneerapplication.model.CategoriesModelItem
import com.amey.pioneerapplication.repository.RemoteDataSource
import com.amey.pioneerapplication.repository.RemoteDataSource.Companion.newInstance
import com.amey.pioneerapplication.utils.AppConstants

class CategoriesViewModel : ViewModel() {
    var remoteDataSource: RemoteDataSource? = null
    var categories: LiveData<List<CategoriesModelItem?>>? = null

    // TODO: Implement the ViewModel
    suspend fun init() {
        if (categories != null) return
        remoteDataSource = newInstance()
        assert(remoteDataSource != null)
        categories =
            remoteDataSource!!.getCategories()
    }

}